wasmplugin/plugin.go

```diff
@@ -236,6 +236,14 @@ func (ctx *httpContext) OnHttpRequestHeaders(numHeaders int, endOfStream bool) t
 	ctx.metrics.CountTX()
 
 	authority, err := proxywasm.GetHttpRequestHeader(":authority")
```

Comment:
It looks like `err` gets possibly ignored now. Let's move `if err != nil` up here with the lines from 269-270 and remove `if err == nil`. This will follow the same pattern as the other properties